src/stores/user-store.js

```diff
@@ -52,21 +53,78 @@ class UserStore extends EventEmitter {
         }
     }
 
+    _findUser (userInfo) {
+        const hasMatchedFields = (item, name) => item[name].toLowerCase().includes(userInfo[name].trim().toLowerCase());
+
+        return this._users.filter(anyUser => {
+            return hasMatchedFields(anyUser, 'name') &&
+                   hasMatchedFields(anyUser, 'phone') &&
+                   hasMatchedFields(anyUser, 'email') &&
+                   hasMatchedFields(anyUser, 'website');
+        });
+    }
+
+    _defineSearchFields () {
+        return {
+            name:    !!this._searchedUser.name.trim(),
+            phone:   !!this._searchedUser.phone.trim(),
+            email:   !!this._searchedUser.email.trim(),
+            website: !!this._searchedUser.website.trim()
+        };
+    }
+
+    _checkSearchModeAndEmit (emit) {
```

Comment:
I would better to name this method e.g. `executeSearch` and leave only 
```
const _foundUsers = this._findUser(this._searchedUser);

this.emit(EVENT_TYPE.usersFound, _foundUsers, this._defineSearchFields());
``` 
in it.

Outside I will use the folowing way (e.g. for `addNewUser`)
```js
if (this.searchMode)
    this.executeSearch();
else
   this.emit(EVENT_TYPE.userAdded);
```
It provide more code but our method `_checkSearchModeAndEmit ` would be more atomic and undestandable.

This way I can use searchMode getter to get info about `this._searchedUser instanceof UserInfo` easier